### PR TITLE
Removes code duplication and adds new unit tests.

### PR DESCRIPTION
**Pull request by diptorupd**
- [X] Have you provided a meaningful PR description?
    - Removes duplicate code inside codegen module.
    - Adds a unit test to make sure a compiler warning is generated
      when the `inline_threshold` is set to >=3
- [X] Have you added a test, reproducer or referred to an issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: numba_dpex/core/codegen.py [hunk 1]
```diff
@@ -55,44 +57,22 @@ def _optimize_final_module(self):
...
-        else:
-            if (
-                hasattr(self, "_inline_threshold")
-                and self._inline_threshold > 0
-                and self._inline_threshold <= 3
-            ):
-                logging.warning(
-                    "Setting INLINE_THRESHOLD leads to very aggressive "
-                    + "optimizations that may produce incorrect binary."
+        if hasattr(self, "_inline_threshold") and self._inline_threshold > 0:
```
Comment (ZzEeKkAa) [inline numba_dpex/core/codegen.py:68]:
```suggestion
        pmb.inlining_threshold = self.inline_threshold
```
why not to set it unconditionally? We can warn in the `@inline_threshold.setter`

**Review by ZzEeKkAa — APPROVED** [review]
LGMT! Thank you. Just one nitpicking

File: numba_dpex/core/codegen.py (hunk 1 above)
Comment (diptorupd) [reply numba_dpex/core/codegen.py]: Makes sense. Done

**Review by ZzEeKkAa — APPROVED** [review]
LGTM! Thank you!

**Review by ZzEeKkAa — APPROVED** [review]

**Pull request merged**